olive/evaluator/olive_evaluator.py

```diff
@@ -614,25 +620,30 @@ def _inference(
 
         preds = []
         targets = []
+        logits = []
         device = PyTorchEvaluator._device_string_to_torch_device(device)
         if device:
             session.to(device)
         for input_data, labels in dataloader:
             input_data = tensor_data_to_device(input_data, device)
             result = session(**input_data) if isinstance(input_data, dict) else session(input_data)
+            if not isinstance(result, torch.Tensor) and getattr(result, "logits", None) is not None:
+                result = result.logits
```

Comment:
a lot of our post funcs are already implement assuming the original state of `result`. I think it might be better to save the logit separately and not overwrite the original result.

Reply:
This condition can be used at line 637 directly.